### PR TITLE
Update Autorest C#

### PR DESCRIPTION
**Pull request by pakrym**
Turned to many properties readonly in the last update, properties for pure input models are back to being readonly.

### CODE REVIEW

**Review by annelo-msft — APPROVED** [review]
LGTM re: FR

File: sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ internal AnalyzeRequest(string text, AnalyzerName? analyzer, TokenizerName? toke
         }
 
         /// <summary> The text to break into tokens. </summary>
-        public string Text { get; set; }
+        public string Text { get; }
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: Why was this changed? It's an input like `Analyzer` and others, which are still read-write. This seems inconsistent. What's the decision point(s) for what's read-only vs. read-write.

Comment (pakrym) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: Because this is a purely input model and Analyzer is input-output, so it can be both sent an received from the server.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: Looking at the swagger, I don't see how you make that determination. The whole model is input. The only difference is that "text" is required, but could still be changed before being sent across the wire.

Comment (pakrym) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: We categorize models based on whether they are a parameter to operations (input) result of the operations (output) or both (mixes).

These rules also apply transitively, so if model A has a property of type B and it's an input B becomes input too.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: I don't think that logic is correct. Yes, `Analyzer` is input/output but it's not on *this* model. This model is purely input. All properties should be read-write. Lots of types in Search are used on both input-only, output-only, and roundtrip models. Properties of those types on models should not necessarily reflect that. How the model is used is what matters.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: /cc @KrzysztofCwalina @tg-msft

Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: Here's my understanding:
1. **Required** properties on **Input** only model types are provided via **.ctor**.
2. **Optional** properties on **Input** only model types are provided via **setters**.
3. **Required and Optional** properties on **Input/Output** model types should have **setters.**
4. **Required** properties on **Input/Output** model types are provided via **.ctors** as well.

I think it matches what we have in this PR.  The guidelines don't capture a lot of this explicitly though.  Relevant sections:
- [Input model types](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-parameters) (which doesn't mention .ctors)
- [Output model types](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-return-model-public-getters)

Comment (KrzysztofCwalina) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: The only thing I am not sure about is #3. Why would required properties be settable? These would already be passed in the ctor.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: I brought that up in the previous PR to update autorest.csharp and its to that inconsistency I refer. If required properties are read-write for I/O models, why not input models? Should either be read-only across the board, or read-write. To solve the validation, we could do something like this in the setters for required properties:
```csharp
set => _backingField = value ?? throw new InvalidOperationException(); // or something
```
But I think making them read-only across all models - taking them in as ctor parameters - is best. It's obvious at compile time the value is required and is consistent with existing packages we've been shipping for a while.

Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: So we can have both:

```C#
Model m = new Model("id", name: "foo");
await client.CreateAsync(m);
```

and 

```C#
Model m = await client.GetAsync("id");
m.Name = "bar";
await client.UpdateAsync(m);
```

Comment (KrzysztofCwalina) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: OK, yes, if the type is designed for round tripping, then the property should be settable. If there are no round trip scenarios, the API designer should probably remove the setter with generator option.

Comment (KrzysztofCwalina) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/AnalyzeRequest.cs:44]: Rephrasing Ted's guidelines a bit to make one thing more explcit:

1. Required properties on input-only models: .ctor + getter.
2. Optional properties: setter + getter.
3. Required properties on input-output models: .ctors + setter + getter;

**Comment by pakrym** [comment]
OK, seems we are all on the same page in terms of guidelines, merging to unblock partners.
